### PR TITLE
refactor: replace char_u variables and functions with char

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

File: src/nvim/runtime.c [hunk 1]
```diff
@@ -42,25 +42,25 @@ void ex_runtime(exarg_T *eap)
 
   if (STRNCMP(arg, "START", len) == 0) {
     flags += DIP_START + DIP_NORTP;
-    arg = skipwhite(arg + len);
+    arg = (char_u *)skipwhite((char *)arg + len);
```
Comment (bfredl) [inline src/nvim/runtime.c:45]: change type of arg to char instead?

File: src/nvim/getchar.c [hunk 2]
```diff
@@ -2874,15 +2874,15 @@ void set_maparg_lhs_rhs(const char *const orig_lhs, const size_t orig_lhs_len,
...
+                                     cpo_flags);
   mapargs->lhs_len = STRLEN(replaced);
   STRLCPY(mapargs->lhs, replaced, sizeof(mapargs->lhs));
   if (did_simplify) {
-    replaced = (char *)replace_termcodes((char_u *)orig_lhs, orig_lhs_len, (char_u **)&alt_lhs_buf,
-                                         flags | REPTERM_NO_SIMPLIFY,
-                                         NULL, cpo_flags);
+    replaced = replace_termcodes(orig_lhs, orig_lhs_len, &alt_lhs_buf,
+                                 flags | REPTERM_NO_SIMPLIFY,
+                                 NULL, cpo_flags);
```
Comment (zeertzjq) [inline src/nvim/getchar.c:2885]: I think you can join some lines here?

File: src/nvim/getchar.c [hunk 3]
```diff
@@ -2874,15 +2874,14 @@ void set_maparg_lhs_rhs(const char *const orig_lhs, const size_t orig_lhs_len,
   // If something like <C-H> is simplified to 0x08 then mark it as simplified.
   bool did_simplify = false;
   const int flags = REPTERM_FROM_PART | REPTERM_DO_LT;
-  char *replaced = (char *)replace_termcodes((char_u *)orig_lhs, orig_lhs_len, (char_u **)&lhs_buf,
-                                             flags, &did_simplify,
-                                             cpo_flags);
+  char *replaced = replace_termcodes(orig_lhs, orig_lhs_len, &lhs_buf,
+                                     flags, &did_simplify,
+                                     cpo_flags);
```
Comment (zeertzjq) [inline src/nvim/getchar.c:2879]: You can also join some lines here.

File: src/nvim/option.c [hunk 4]
```diff
@@ -4978,7 +4980,7 @@ int get_option_value(const char *name, long *numval, char_u **stringval, int opt
       return -2;
     }
     if (stringval != NULL) {
-      *stringval = vim_strsave(*(char_u **)(varp));
+      *stringval = (char *)vim_strsave(*(char_u **)(varp));
```
Comment (bfredl) [inline src/nvim/option.c:4983]: `xstrdup`

**Comment by bfredl** [comment]
LGTM, squash?

**Pull request merged**